### DISCUSSION TITLE
Dynamic favicon

### DISCUSSION THREAD

**Discussion by jbdemonte** [vercel/next.js#59801] [Help]
### Summary

display a favicon / icon from a database entry

### Additional information

Using next 14 and app router, we are providing a dashboard where a user can upload an icon / favicon to be used on his pages.

Using pages, I would have use <Head> but in the app router, I don't find how to display the icon in the head, while loading the data from firebase on client side (use client)

**Comment by trobinson-sx** [comment]
You can either export a `generateMetadata` function to dynamically return a different URL:

```ts
export function generateMetadata() {
  return {
    icons: [{ rel: "icon", url: "/favicon.ico" }]
  }
}
```

https://nextjs.org/docs/app/api-reference/functions/generate-metadata#generatemetadata-function

~Or dynamically generate a favicon response with a file at `app/favicon.tsx`:~

https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons#generate-icons-using-code-js-ts-tsx

**Reply by Arctomachine** [reply]
From this page:

> You cannot generate a favicon icon. Use [icon](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons#icon) or a [favicon.ico](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons#favicon) file instead.
